.github/workflows/e2e.yaml

```diff
@@ -68,58 +66,48 @@ jobs:
       - uses: subosito/flutter-action@v2
         with:
           channel: stable
-      - name: Build and install the appxbundle for testing
+      - name: Clean up leftovers from previous runs
         shell: powershell
         run: |
-          # If there is any leftovers from a previous run.
           $package=$(Get-AppxPackage | Where-Object PackageFullName -like *${{ env.appID }}*)
           If ( $package -ne $null -And $package -ne '' ) {
             Remove-AppxPackage $package
           }
-          .\e2e\pre-e2e-testing.ps1 -AppID ${{ env.appID }} -RootfsX64 ${{ env.rootfs64 }}
-      - name: Run tests
+          exit 0 # Always suceed
+      - name: Build and install the appxbundle for testing
         shell: powershell
         run: |
-          $higherSession=0
-          $count = 0
-          While ($higherSession -lt 2) {
-            Sleep 5
-            $count = $count + 1
-            If ($count -gt 19) {
-              Write-Error "RDP never connected"
-              Exit (-1)
-            }
-
-            $sessions=$(C:\SysinternalsSuite\logonsessions.exe -nobanner | Select-String Session: | Sort-Object -Descending | Get-Unique)
-            echo $sessions
-            If ( $sessions -ne $null -and $sessions.length -gt 0 ) {
-              $higherSession=$( $sessions )[0].ToString()[-1]
+          .\e2e\pre-e2e-testing.ps1 -AppID ${{ env.appID }} -RootfsX64 ${{ env.rootfs64 }}
+      - name: Run tests
+        uses: ./.github/actions/session_nonzero/
+        with:
+          exec: |
+            $env:WSL_UTF8=1
+            $env:LAUNCHER_REPO_ROOT=$PWD 
+            wsl.exe --version
+            cd .\e2e
+            go test .\test-runner -timeout 15m -run TestBasicSetup --distro-name ${{ env.distroName }} --launcher-name ${{ env.launcher}}
+            If ($LastExitCode) {

```

Comment:
Do you mind commenting on that for a non powershell expert? :p (it doesn’t exit with the last command value?)

Reply:
Scripts do indeed not exit with the last exit code. See this example:

```powershell
PS > cat .\deleteme.ps1
python.exe -c "exit(42)"
PS > powershell.exe -ExecutionPolicy Bypass -File .\deleteme.ps1
PS > echo $LASTEXITCODE
0
```

Reply:
Hving said this, those three lines of code can be (and now have been) simplified.

Reply:
My point was: do we really need this exit then? if `go test` returns 1, the script itself won’t return 1 (returning last execution status, as bash does)?